src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -964,6 +964,7 @@ def get_image_features(
         output_hidden_states: Optional[bool] = None,
         return_dict: Optional[bool] = None,
         return_projected: Optional[bool] = True,
```

Comment:
Sorry about the confusion, the `return_projected` argument is left over from previous commits and was replaced with the `return_base_image_embeds` argument in `OwlViTModel.forward()` later on. 

- When `return_base_image_embeds=False`, unmodified OwlViTModel (CLIP) embeddings are returned. 
- When `return_base_image_embeds=True`, post_layernorm is applied to the last hidden state instead of the pooled output and no visual projection is applied. We use this setting to embed images within OwlViTForObjectDetection

```suggestion
```

Reply:
so when `return_base_image_embeds=False` will we return `vision_outputs[1]`  or the full object `visual_outputs` itself?

Reply:
OwlViTModel.get_image_features() only needs to return the image embeddings so we'd do:

```
pooled_output = vision_outputs[1]
image_features = self.visual_projection(pooled_output)

if return_base_image_embeds:
    last_hidden_state = vision_outputs[0]
    image_features = self.vision_model.post_layernorm(last_hidden_state)

return image_features
```